File: example/periphery/gpio/python/test_gpio_read.py
from lockzhiner_vision_module.periphery import GPIO0A0
from lockzhiner_vision_module.periphery import GPIOMode
from lockzhiner_vision_module.periphery import GPIOState

if __name__ == "__main__":
    gpio = GPIO0A0()
    
    if gpio.config(GPIOMode.IN) is False:
        print("Failed to config gpio mode")
        exit(1)
        
    state = gpio.read()
    if state == GPIOState.ERROR:
        print("Failed to read gpio mode")
        exit(1)
        
    print(f"state is {state}")

File: example/periphery/gpio/cpp/test_gpio_read.cc
#include <lockzhiner_vision_module/periphery/gpio/gpio.h>

#include <iostream>
#include <thread>

int main() {
  lockzhiner_vision_module::periphery::GPIO0A0 gpio;

  if (!gpio.Config(lockzhiner_vision_module::periphery::GPIOMode::IN)) {
    std::cout << "Failed to config gpio mode" << std::endl;
    return 1;
  }

  lockzhiner_vision_module::periphery::GPIOState state;
  if (!gpio.Read(state)) {
    std::cout << "Failed to read gpio mode" << std::endl;
    return 1;
  }

  std::cout << "state is " << static_cast<uint8_t>(state) << std::endl;
  return 0;
}